lib/basket.rb

```diff
@@ -0,0 +1,27 @@
+require_relative 'menu'
+
+class Basket
+    attr_reader :basket, :menu, :total
+    def initialize(menu = Menu.new)
+        @menu = menu
+        @basket = {}
+    end
+
+    def add(item, quantity)
+        @menu.menu.select { |dish, price| @basket[item] = quantity if dish == item }
+    end
+
+    def show_basket
+        basket = []
+        @basket.each { |item, quantity| basket << "#{item} x #{quantity}" }
+        basket.join(", ")
+    end
+
+    def checkout
+        checkout = []
```

Comment:
similar array variable in the show_basket method. Could use same instant variable to get total.